demo/api/src/app.module.ts

```diff
@@ -81,19 +81,23 @@ export class AppModule {
                         return user.domains.includes(requestScope.domain);
                     },
                 }),
-                UserPermissionsModule.forRootAsync({
-                    useFactory: (userService: UserService) => ({
-                        availablePermissions: ["news", "products"],
-                        availableContentScopes: [
-                            { domain: "main", language: "de" },
-                            { domain: "main", language: "en" },
-                            { domain: "secondary", language: "en" },
-                        ],
-                        userService,
-                    }),
-                    inject: [UserService],
-                    imports: [AuthModule],
-                }),
+                UserPermissionsModule.forRootAsync(
+                    {
+                        useFactory: () => ({
+                            availablePermissions: ["news", "products"],
+                            availableContentScopes: [
+                                { domain: "main", language: "de" },
+                                { domain: "main", language: "en" },
+                                { domain: "secondary", language: "en" },
+                            ],
+                        }),
+                    },
+                    {
+                        useFactory: (userService: UserService) => userService,
+                        inject: [UserService],
+                        imports: [AuthModule],
+                    },
+                ),
```

Comment:
I've never seen this pattern in Nest before 🤔

Reply:
https://github.com/nestjsplus/massive-cats/blob/master/src/app.module.ts

Reply:
Or here in a module provided by nest themselves: https://github.com/nestjs/bull/blob/c4f59c605082b1080c7d3b670a72024d462546a7/packages/bull/lib/bull.module.ts#L101